### PR TITLE
#61 update etl

### LINKED ISSUE #61: Update extract and load
**Description:**
Enhance the extract and load processes to include robust logging and error handling, ensuring data extraction and loading are traceable and resilient to failures.

**Outcome:**
An updated extract and load process that logs key activities and errors, providing better visibility into the data pipeline and handling any failures gracefully.

**User Story:**
As a data engineer, I need to include logging and error handling in the extract and load processes so I can track data flow, identify issues, and ensure the pipeline continues running smoothly.

**Design:**
    **Logging:**
        Add logging at key stages of the extract and load processes, including the start and end of each stage, the number of records processed, and any significant operations.
        Log any warnings or errors encountered during extraction or loading for later debugging.
    **Error Handling:**
        Implement error handling mechanisms to catch and log failures (e.g., network issues during extraction or write failures during loading).

### PR DESCRIPTION
**Pull request by qrafiq1**
Added error handling and logging to extract and load. Added extra tests for error raising.

Closes #61

### CODE REVIEW

File: pipeline/test_load.py [hunk 1]
```diff
@@ -98,3 +98,107 @@ def test_insert_into_earthquake_valid(mock_connection, mock_cursor, valid_earthq
...
+    """Test insert_into_earthquake with empty data"""
+    insert_into_earthquake(mock_connection, mock_cursor, [])
+    mock_cursor.fetchone.assert_not_called()
+    mock_connection.commit.assert_not_called()
+
+
+@patch('psycopg2.connect')
+def test_get_connection_success(mock_connect):
+    """Test get_connection with a successful connection."""
+    # Mock the connection object returned by psycopg2.connect
```
Comment (S1mpySloth) [inline pipeline/test_load.py:113]: Can likely remove these comments now, the docstring and variable names etc. all serve as sufficient documentation :)

File: pipeline/test_extract.py [hunk 2]
```diff
@@ -75,7 +77,7 @@ def test_get_data_no_recent_data(mock_get):
 
 
 @patch('extract.requests.get')
-def test_get_data_missing_fields(mock_get):
+def test_get_data_missing_fields_valid(mock_get):
```
Comment (S1mpySloth) [inline pipeline/test_extract.py:80]: Should these tests have docstrings, to serve as extra documentation?

**Review by S1mpySloth — APPROVED** [review]
Looks great, just a couple 'best-practice' tweaks from me, otherwise, good stuff!

**Review by jiuliangut — APPROVED** [review]
great job on the error handling, nice and thorough

**Review by Fahi28 — APPROVED** [review]
nice logging and error handling!

**Pull request merged**